batchflow/tests/pipeline_init_model_test.py

```diff
@@ -0,0 +1,55 @@
+""" Tests for pipeline init_model action """
+# pylint: disable=import-error, no-name-in-module
+import pytest 
+
+from batchflow import DatasetIndex, Pipeline, Batch, C
+from batchflow.models import BaseModel
+from batchflow.model_dir import NonInitializedModel
+
+@pytest.fixture
+def dataset():
+    """ Toy dataset """
+    index = DatasetIndex(100)
+    return Dataset(index, Batch)
+
+@pytest.mark.parametrize('use_algebra', [True, False])
+@pytest.mark.parametrize('config', [dict(mode='static', base_class=BaseModel, name='model', config={}),
+                                    dict(mode='dynamic', base_class=BaseModel, name='model', config={})])
+def test_single_model(use_algebra, config, dataset):
+    """ Verifies that the model in init_model is initialized as expected with config provided, i.e.
+    pipeline contains the model with given name belongs to given class.
+    Also veriying methods of linking config to the pipeline: via pipeline algebra and manually. """
+    if not use_algebra:
+        pipeline = dataset.pipeline(config).init_model(C('mode'), C('base_class'), C('name'), C('config'))
+    else:
+        pipeline = dataset.pipeline().init_model(C('mode'), C('class'), C('base_name'), C('config')) << config
+    pipeline.next_batch(1)
+
+    if config['mode'] == 'static': 
+        assert type(pipeline.m(config['name']) == config['class']
+    else:
+        assert type(pipeline.m(config['name']) == NonInitializedModel
```

Comment:
`is` or `isinstance` fits the purpose
if a model is NonInitializedModel then it was not initialized apparently